### PR TITLE
Preseve stack trace in nodes during fx.Transform

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82670

### CODE REVIEW

File: torch/fx/interpreter.py [hunk 1]
```diff
@@ -148,6 +149,8 @@ def run_node(self, n : Node) -> Any:
         Returns:
             Any: The result of executing ``n``
         """
+        self.current_node = n
```
Comment (ezyang) [inline torch/fx/interpreter.py:152]: I'd prefer this to be dynamically scoped: so you save the current value of `current_node`, set it, and then on return (finally) you restore to old value

Comment (ezyang) [reply torch/fx/interpreter.py:152]: This is mostly helpful if you have some reentrant execution of the same interpreter. Which hopefully is pretty unlikely...

**Review by ezyang — APPROVED** [review]
okey dokey; manually written passes are going to have to be adjusted too though :(

@Chillee it's easier to hide the boilerplate if we do retracing lol. I swear people are going to keep forgetting to propagate stack traces. Maybe we need to make some modifications to the API people do node iteration over to make it easier to not forget to do this.

File: torch/fx/interpreter.py (hunk 1 above)
Comment (SherlockNoMad) [reply torch/fx/interpreter.py]: Updated per request.

**Pull request closed without merging**